symplyphysics/laws/optics/interference_films_from_angle_refraction_refractive_index_wavelength.py

```diff
@@ -0,0 +1,57 @@
+from sympy import (Eq, solve, cos)
+from symplyphysics import (
+    units,
+    Quantity,
+    Symbol,
+    print_expression,
+    validate_input,
+    validate_output,
+    dimensionless,
+    angle_type
+)
+
+# Description
+## Interference in thin films is a phenomenon that occurs as a result of the separation of a light beam when
+## reflected from the upper and lower boundaries of a thin film. As a result, there are two light waves that
+## can interfere.
+## The order of interference indicates the number of whole wavelengths. When the difference in the path of the
+## beam reflected from the upper boundary of the film and the beam reflected from the lower boundary of the
+## film is equal to an integer wave "n", then these two waves enter the observation point with the same phases
+## and demonstrate interference.
+## The angle of refraction of light is the angle between the ray passing through the surface and the normal to the surface.
+
+## Law is: h = k * L / (2 * n * cos(O)), where
+## h - film thickness,
```

Comment:
What is this thickness that we've got here? What does it mean?

Reply:
Yes, but what does this thickness mean? What happens if it is 1 centimeter?

Reply:
> Yes, but what does this thickness mean? What happens if it is 1 centimeter?

Depends on other values. The interference pattern may not change at all, it may become the exact opposite (maxima will change with minima), or it may change arbitrarily.

Reply:
> Yes, but what does this thickness mean? What happens if it is 1 centimeter?

At the moment, I completely do not understand the essence of the remark and what needs to be corrected.

Reply:
The reason for this remark is that law should have any usability for the users. What's the point having a law, that gives back some arbitrary thickness? You could as well just generate some random output.
You have some calculations that give to user some thickness. That's why I am asking, what is this thickness? Why is it important?

Reply:
> The reason for this remark is that law should have any usability for the users. What's the point having a law, that gives back some arbitrary thickness? You could as well just generate some random output.
> You have some calculations that give to user some thickness. That's why I am asking, what is this thickness? Why is it important?

With the interference order equal to 1, with other parameters set, this thickness will be minimal in order to see some kind of interference pattern. That is, for a fixed material, the angle of refraction and wavelength, it is possible to determine the minimum thickness of the film so that an interference pattern is observed.

Reply:
Are you sure? I see something about minimum of interference is visible, when number of half waves is even. Shouldn't some kind of interference pattern be visible with any order of interference number?

Reply:
> Are you sure? I see something about minimum of interference is visible, when number of half waves is even. Shouldn't some kind of interference pattern be visible with any order of interference number?

There is a condition for the minimum interference pattern at the observation point at infinity (4.24b):
http://fn.bmstu.ru/data-physics/library/physbook/tom4/ch4/texthtml/ch4_4.htm

We will set all the parameters except the thickness of the film. Then, for some film values that are less than the minimum thickness value, the equality of 4.24b simply will not be fulfilled. If the interference order was set to 1, then for large parameters this equality will not be fulfilled for thickness values less than the minimum. If the interference order is arbitrary and equal to "m", then condition 4.24 will not be satisfied for all values of the order from m to infinity with a width less than the minimum.

That's what I meant.

Reply:
It is not a minimal thickness to see interference pattern. It is thickness that gives you interference minimum at a certain point on the interference pattern. So you should update law name and description accordingly.

Reply:
> It is not a minimal thickness to see interference pattern. It is thickness that gives you interference minimum at a certain point on the interference pattern. So you should update law name and description accordingly.

In many tasks that I have seen, they ask me to find the minimum film thickness at which a maximum or minimum intensity will be observed for a given wavelength and angle of refraction. I do not understand any other explanation of the meaning of thickness, other than the one I described above. Therefore, I do not understand how I should change the name of the law and its description.

Reply:
You are messing two things: interference minimum and minumal thickness. With your law you can find thickness that results in interference minimum at a certain point. It does not necessarily results in minimal thickness. For example by changing the order of interference you can change the resulting thickness so it is no longer minimal.